internal/client/networking.go

```diff
@@ -79,13 +108,11 @@ func (n *networkingClient) ReserveIP(
 		)
 	}
 
-	result, err := n.client.Prism().WaitForTaskCompletion(
-		ctx,
+	result, err := n.client.Prism().GetTaskData(
 		*responseData.ExtId,
-		opts.WaitForTaskCompletionOpts,
 	)
 	if err != nil {
-		return nil, fmt.Errorf("failed to wait for task completion: %w", err)
+		return nil, fmt.Errorf("task has not successfully completed: %w", err)
```

Comment:
Is this case is an error of retrieving task data or does i signal that the task has not completed?

Reply:
Good catch. Updated to only wrap the error if we fail to check the status, otherwise return the error directly without wrapping.